src/transformers/models/superpoint/modeling_superpoint.py

```diff
@@ -0,0 +1,577 @@
+        keypoints, scores = self.remove_borders(keypoints, scores, self.border_removal_distance, height * 8, width * 8)
+
+        # Keep the k keypoints with highest score
+        if self.max_keypoints >= 0:
+            keypoints, scores = self.top_k_keypoints(keypoints, scores, self.max_keypoints)
+
+        # Convert (y, x) to (x, y)
+        keypoints = torch.flip(keypoints, [1]).float()
+
+        return keypoints, scores
+
+    @staticmethod
+    def simple_nms(scores: torch.Tensor, nms_radius: int) -> torch.Tensor:
+        if nms_radius < 0:
+            raise ValueError("Expected positive values for nms_radius")
+
+        def max_pool(x):
+            return torch.nn.functional.max_pool2d(x, kernel_size=nms_radius * 2 + 1, stride=1, padding=nms_radius)
+
+        zeros = torch.zeros_like(scores)
+        max_mask = scores == max_pool(scores)
+        for _ in range(2):
+            supp_mask = max_pool(max_mask.float()) > 0
+            supp_scores = torch.where(supp_mask, zeros, scores)
+            new_max_mask = supp_scores == max_pool(supp_scores)
+            max_mask = max_mask | (new_max_mask & (~supp_mask))
+        return torch.where(max_mask, scores, zeros)
+
+    @staticmethod
+    def remove_borders(
+        keypoints: torch.Tensor, scores: torch.Tensor, border: int, height: int, width: int
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        """Removes keypoints too close to the border"""
+        mask_h = (keypoints[:, 0] >= border) & (keypoints[:, 0] < (height - border))
+        mask_w = (keypoints[:, 1] >= border) & (keypoints[:, 1] < (width - border))
+        mask = mask_h & mask_w
+        return keypoints[mask], scores[mask]
+
+    @staticmethod
+    def top_k_keypoints(keypoints: torch.Tensor, scores: torch.Tensor, k: int) -> Tuple[torch.Tensor, torch.Tensor]:
+        if k >= len(keypoints):
+            return keypoints, scores
+        scores, indices = torch.topk(scores, k, dim=0)
+        return keypoints[indices], scores
+
+
+class SuperPointDescriptorDecoder(nn.Module):
+    """
+    The SuperPointDescriptorDecoder uses the outputs of both the SuperPointEncoder and the
+    SuperPointInterestPointDecoder to compute the descriptors at the keypoints locations.
+
+    The descriptors are first computed by a convolutional layer, then normalized to have a norm of 1. The descriptors
+    are then interpolated at the keypoints locations.
+    """
+
+    def __init__(self, config: SuperPointConfig) -> None:
+        super().__init__()
+        self.hidden_sizes = config.hidden_sizes
+        self.descriptor_dim = config.descriptor_dim
+        self.keypoint_threshold = config.keypoint_threshold
+        self.max_keypoints = config.max_keypoints
+        self.nms_radius = config.nms_radius
+        self.border_removal_distance = config.border_removal_distance
+
+        self.relu = nn.ReLU(inplace=True)
+        self.pool = nn.MaxPool2d(kernel_size=2, stride=2)
+        self.convDa = nn.Conv2d(
```

Comment:
We shouldn't have capital letters in var names. What does `Da` and `Db` stand for? Let's replace it with something more explicit

Reply:
Fixed in [this commit](https://github.com/huggingface/transformers/pull/25786/commits/c43052de41132765b2bdfccb863a0ee7aa33a9c4)